e2e/test/scenarios/admin-2/api-keys.cy.spec.ts

```diff
@@ -0,0 +1,291 @@
+import type { CyHttpMessages } from "cypress/types/net-stubbing";
+
+import { popover, restore } from "e2e/support/helpers";
+import type { ApiKey } from "metabase-types/api";
+
+const MOCK_ROWS: ApiKey[] = [
+  {
+    name: "Development API Key",
+    id: 1,
+    group_id: 1,
+    group_name: "All Users",
+    creator_id: 1,
+    masked_key: "asdfasdfa",
+    created_at: "2010 Aug 10",
+    updated_at: "2010 Aug 10",
+  },
+  {
+    name: "Production API Key",
+    id: 2,
+    group_id: 2,
+    group_name: "Administrators",
+    creator_id: 1,
+    masked_key: "asdfasdfa",
+    created_at: "2010 Aug 10",
+    updated_at: "2010 Aug 10",
+  },
+  {
+    name: "Personal API Key",
+    id: 3,
+    group_id: 3,
+    group_name: "collection",
+    creator_id: 1,
+    masked_key: "asdfasdfa",
+    created_at: "2010 Aug 10",
+    updated_at: "2010 Aug 10",
+  },
+];
+
+const MOCK_GROUPS = [
+  // NOTE: this is only here temporarily (remove when backend is working)
+  {
+    id: 2,
+    name: "Administrators",
+    member_count: 1,
+  },
+  {
+    id: 1,
+    name: "All Users",
+    member_count: 9,
+  },
+  {
+    id: 3,
+    name: "collection",
+    member_count: 4,
+  },
+  {
+    id: 4,
+    name: "data",
+    member_count: 2,
+  },
+  {
+    id: 6,
+    name: "nosql",
+    member_count: 1,
+  },
+  {
+    id: 5,
+    name: "readonly",
+    member_count: 1,
+  },
+];
+
+const getRequestId = (req: CyHttpMessages.IncomingHttpRequest) =>
+  parseInt(req.url.match(/api-key\/(\d+)/)?.[1] ?? "", 10);
+
+describe("scenarios > admin > settings > API keys", () => {
+  // TODO: replace intercepts below with actual requests to test backend
+  let mockRows: ApiKey[] = [];
+
+  beforeEach(() => {
+    restore();
+    mockRows = [...MOCK_ROWS];
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    cy.signInAsAdmin();
```

Comment:
🤔 another reason for a separate PR to enable typescript, I have a pretty strong aversion to using // @ts-ignore.

 I think we need to define types for our plugins and tasks before enabling typescript specs
 
 https://docs.cypress.io/guides/tooling/typescript-support#Types-for-plugins

Reply:
Added here, thanks: https://github.com/metabase/metabase/pull/36958